src/main/java/org/gaul/s3proxy/S3ProxyHandler.java

```diff
@@ -1315,13 +1321,12 @@ private void handleBlobList(HttpServletRequest request,
         }
         options.maxResults(maxKeys);
 
-        response.setCharacterEncoding("UTF-8");
-
         PageSet<? extends StorageMetadata> set = blobStore.list(containerName,
                 options);
 
+        response.setCharacterEncoding(UTF_8);
```

Comment:
Should we set the character encoding next to the content type for consistency?

Reply:
Sadly no, the encoding is permanently set once the writer is generated, with any `setCharacterEncoding` calls effectively ignored, so I had to set it immediately before instead. Caused me to scratch my head for a while did that.

Reply:
Thanks for investigating this!  This makes sense that the `Writer` would get some state set during construction.